api/v1/verticadb_webhook.go

```diff
@@ -1625,31 +1633,45 @@ func (v *VerticaDB) checkImmutableSubclusterInSandbox(oldObj *VerticaDB, allErrs
 			continue
 		}
 	}
+	return allErrs
+}
 
-	// find subclusters that are sandboxed in old vdb but removed in new vdb
+// checkSandboxSubclustersRemoved checks subclusters that are sandboxed in old vdb but removed in new vdb
+func (v *VerticaDB) checkSandboxSubclustersRemoved(allErrs field.ErrorList, oldObj *VerticaDB, oldScIndexMap map[string]int,
+	oldScMap, newScMap map[string]*Subcluster, path *field.Path) field.ErrorList {
 	oldScInSandbox := oldObj.GenSubclusterSandboxMap()
```

Comment:
We should modify this to: newScInSandbox := v.GenSubclusterSandboxMap(). Then in line 1647, verify the removed sc isn't in newScInSandbox. The check in line 1650(if v.GetSandbox(sb) == nil) is not accurate since the sc could be in a new sandbox of new vdb.